ledger/txtail.go

```diff
@@ -173,9 +179,13 @@ func (t *txTail) loadFromDisk(l ledgerForTracker, dbRound basics.Round) error {
 
 	// add all the entries in roundsLastValids to their corresponding map entry in t.lastValid
 	for lastValid, list := range lastValid {
-		lastValueMap := make(map[transactions.Txid]struct{}, len(list))
-		for _, id := range list {
-			lastValueMap[id] = struct{}{}
+		lastValueMap := make(map[transactions.Txid]int16, len(list))
+		for _, entry := range list {
+			if lastValid < entry.rnd {
```

Comment:
how could this ever happen?

Reply:
It should not but these values come from DB and I thought it is better to check them just in case.